SYCL/Matrix/joint_matrix_tensorcore.cpp

```diff
@@ -1,6 +1,9 @@
-// REQUIRES: gpu, cuda
+// REQUIRES: cuda
 
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -Xsycl-target-backend --cuda-gpu-arch=sm_80 -DSYCL_EXT_ONEAPI_MATRIX=3  %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -Xsycl-target-backend --cuda-gpu-arch=sm_80 -DSYCL_EXT_ONEAPI_MATRIX=3 %s -o %t.out
+// TODO: Currently the CI does not have a sm_80 capable machine. Enable the test
+// execution once it does.
+// RUNx: %t.out
```

Comment:
Instead of relying on CI, could the test check the SM version of the device during execution and skip if it isn't SM80?

Additionally, does all of this test require SM80 or is it just a subset? If it is just a subset, it would be better to split it up so CI can at least test the parts that are expected to work with the current hardware.

Reply:
Thank you! Remind me, when does the CUDA backend fail if the device does not have the capabilities required for a kernel? Is it during build or is it when the specific kernel is enqueued?

If it's the former, the test may still fail despite the checks as the kernels may be put in the same device code module. There are a handful of ways to work around it:

1. Split the test into separate test files. Optionally they could share a common header.
2. Use macros in this test and compile and run each variant.
3. Add `-fsycl-device-code-split=per_kernel`. This will result in separate device code for each kernel, which may be a little aggressive here.

Reply:
> Thank you! Remind me, when does the CUDA backend fail if the device does not have the capabilities required for a kernel? Is it during build or is it when the specific kernel is enqueued?
> 

It's the latter. I've ran the test on a lower SM arch than the tests support and it will currently skip the tests and pass.

Reply:
Thanks for clarifying!